expression/builtin_cast_vec.go

```diff
@@ -834,11 +840,48 @@ func (b *builtinCastDurationAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, res
 }
 
 func (b *builtinCastIntAsDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastIntAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	isUnsigned := mysql.HasUnsignedFlag(b.tp.Flag) || mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
+	nums := buf.Int64s()
+	result.ResizeDecimal(n, false)
+	result.MergeNulls(buf)
+	decs := result.Decimals()
+	sc := b.ctx.GetSessionVars().StmtCtx
+	dec := new(types.MyDecimal)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		*dec = types.MyDecimal{}
+		if !isUnsigned {
+			dec.FromInt(nums[i])
+		} else if b.inUnion && nums[i] < 0 {
```

Comment:
There's a bug fix #13637 involving with `builtinCastIntAsDecimalSig`, maybe we should fix it here after that PR is merged.

Reply:
Fixed.